ui/src/app/components/admin/audit-card-list-view.tsx

```diff
@@ -193,17 +210,115 @@ const AuditActionCard = (props: { action: AuditAction }) => {
   );
 };
 
-export const AuditActionCardListView = (props: { actions: AuditAction[]}) => {
+interface FilterEntry {
+  displayName?: string;
+  isActive: boolean;
+}
+
+const ActionTypeFilter = (props: {
+  activeActionTypes: { [key: string]: FilterEntry },
+  updateFilter: (actionType: string, isActive: boolean) => void }) => {
+
+  const {activeActionTypes, updateFilter} = props;
+
+  const toggleSelectedAction = (actionType) => {
+    updateFilter(actionType, !fp.get(`${actionType}.isActive`)(activeActionTypes));
+  };
+
+  return (
+      <React.Fragment>
+        <div style={{
+          margin: '0',
+          display: 'flex',
+          flexDirection: 'column',
+          textAlign: 'left',
+          border: '1px solid',
+          borderColor: colors.secondary,
+          width: 'fit-content'
+        }}>
+        <div style={{fontWeight: 600, color: colors.accent}}>Action Types</div>
+        {fp.entries(activeActionTypes).map((actionTypeValuePair: [string, FilterEntry], index) =>
+          <CheckBox
+            key={index}
+            id={`${actionTypeValuePair[0]}_active`}
+            onChange={() => toggleSelectedAction(actionTypeValuePair[0])}
+            label={actionTypeValuePair[1].displayName}
+            checked={actionTypeValuePair[1].isActive}
+            style={{margin: '0.25rem'}}/>)}
+        </div>
```

Comment:
This might be a bit cleaner with some additional desctructuring.
Also, keeping the code consistent with lodash may help future development and prevent confusion when reading through the code.

I am not a fan of the double `toPairs` in my solution, but it does keep it in lodash/fp... I will think through it a bit more to see if there is a better solution. 

If you don't want to do that - do take a look at the destructuring bit. It may be a bit cleaner than the array index references. The destructure should work with the native map code as well.

```javascript
fp.flow(
          fp.toPairs,
          fp.toPairs,
          fp.map(([index, [id, entry]]: [number, [string, FilterEntry]]) =>
            <CheckBox
              key={index}
              id={`${id}_active`}
              onChange={() => toggleSelectedAction(id)}
              label={entry.displayName}
              checked={entry.isActive}
              style={{margin: '0.25rem'}}/>)
        )(activeActionTypes)}
```

Reply:
It may be better to change what the structure of `activeActionTypes` looks like in `AuditActionCardListView` to make it easier to consume. That may be better for a future PR